ir/instruction.go

```diff
@@ -106,4 +108,6 @@ type Instruction interface {
 	// isInstruction ensures that only instructions can be assigned to the
 	// instruction.Instruction interface.
 	isInstruction()
+	// Operands returns a mutable list of operands of the instruction.
```

Comment:
Should we return a *mutable* list? If so, the return type should probably be `[]*value.Value`?

Reply:
Why? Shouldn't interface already a pointer?

Reply:
The underlying value of a concrete type implementing the `value.Value` interface is often a pointer, e.g. `*ir.InstAdd`. Also, interface types are often treated as pointers, e.g. compare against `nil`.

However, lets consider the use case where you want to *update* the operands of an instruction to other values, then, the following code would not work if using `[]value.Value`.

See example at https://play.golang.org/p/Vy2BnlC7FkD (using `[]value.Value`) and https://play.golang.org/p/ICR50iHBfs9 (using `[]*value.Value`).

Using `updateOperands` (which uses `[]value.Value`) is not able to update the operands (thus not mutable). Note that the `add` instruction was not updated to use `zeroinitializer`:

Output:
```llvm
define i32 @foo(i32 %x, i32 %y) {
entry:
	%0 = add i32 %x, %y
	ret i32 %0
}

define i32 @main() {
entry:
	%0 = call i32 @foo(i32 30, i32 12)
	ret i32 %0
}
```

Using `updateMutableOperands` (which uses `[]*value.Value`) is able to update the operands (thus mutable). Note that the `add` instruction was updated to use `zeroinitializer`:

Output:
```llvm
define i32 @foo(i32 %x, i32 %y) {
entry:
	%0 = add i32 zeroinitializer, zeroinitializer
	ret i32 %0
}

define i32 @main() {
entry:
	%0 = call i32 @foo(i32 30, i32 12)
	ret i32 %0
}
```

Reply:
oh, I get what you mean. However, we can update the operand directly via the instruction itself. Thus, I can't see the reason that makes `Operand` more complicated(be mutable). So I would update the document string.